src/sgame/sg_bot_util.cpp

```diff
@@ -1647,6 +1648,20 @@ void BotFireWeaponAI( gentity_t *self )
 	}
 }
 
+bool BotChangeClass( gentity_t *self, class_t newClass, vec3_t newOrigin )
+{
+	if ( !G_RoomForClassChange( self, newClass, newOrigin ) )
+	{
+		return true;
```

Comment:
probably should return `false` here.

```suggestion
		return false;
```

The expectation is `if (BotChangeClass(…)) { /* worked */ }`

Reply:
If there is not enough room to change class, return that there's an error. This allows to test directly for `error` instead of checking for `!error`, thus:

```
if ( call_1() ) {
  //error handling
  return;
}
//no problem, continue
if ( call_2() ) {
  //error handling
  return;
}
//no problem, continue
```

instead of:

```
if ( call_1() ) {
  //no problem
  if ( call_2() ) {
    //no problem
  }
  return;
}
//error handling
```

or
```
if ( !call_1() ) {
  //error handling
  return;
}
//no problem, continue
if ( !call_2() ) {
  //error handling
  return;
}
//no problem, continue
```